share/spice/twitch_streams/twitch_streams.js

```diff
@@ -0,0 +1,67 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_twitch_streams = function(api_result){
+
+        if (!api_result || api_result.error) {
+            return Spice.failed('twitch_streams');
+        }
+        
+        var streams = [];
+        if (!($.isArray(api_result.streams))) {
+            streams = [api_result.streams];
+        } else {
+            streams = api_result.streams;
+        }  
+        
+        //Get query, as per documentation
+        var script = $('[src*="/js/spice/twitch_streams/"]')[0],
+            source = $(script).attr("src"),
+            query = source.match(/twitch_streams\/([^\/]+)/)[1],
+            decodedQuery = decodeURIComponent(query);
+        //Replace spaces with +
+        var moreAt = decodedQuery.replace(/ /g,"+");
+        
+        //Push results into an array for display
+        var results = [];
+        
+        for (var i = 0; i < streams.length; i++) {
+            var url = streams[i].channel.url;
+            var displayName = streams[i].channel.display_name;
+            if(streams[i].channel.logo != null){
+                var logo = streams[i].channel.logo;
+            } else {
+                var logo = 'http://static-cdn.jtvnw.net/jtv-static/404_preview-300x300.png';
```

Comment:
Any remote image that is pulled will need to be proxied. You can use the [imageProxy](https://duck.co/duckduckhack/spice_handlebars_helpers#imageproxy) handlebars helper in the template or just add the image proxy url in javascript, [example](https://github.com/duckduckgo/zeroclickinfo-spice/blob/83b2db3d77c206ce62dd0ab0ffcc1d65cbbf2130/share/spice/rx_info/rx_info.js#L86)